SimG4Core/Application/src/OscarMTMasterThread.cc

```diff
@@ -119,14 +104,33 @@ OscarMTMasterThread::~OscarMTMasterThread() {
   }
 }
 
+void OscarMTMasterThread::callConsumes(edm::ConsumesCollector&& iC) const {
+  if (m_hasToken) {
+    return;
+  }
+  if (m_pGeoFromDD4hep) {
+    m_DD4Hep = iC.esConsumes<cms::DDCompactView, IdealGeometryRecord, edm::Transition::BeginRun>();
+  } else {
+    m_DDD = iC.esConsumes<DDCompactView, IdealGeometryRecord, edm::Transition::BeginRun>();
+  }
+  m_PDT = iC.esConsumes<HepPDT::ParticleDataTable, PDTRecord, edm::Transition::BeginRun>();
+  m_hasToken = true;
+}
+
 void OscarMTMasterThread::beginRun(const edm::EventSetup& iSetup) const {
   std::lock_guard<std::mutex> lk(m_protectMutex);
-
   std::unique_lock<std::mutex> lk2(m_threadMutex);
+  edm::LogVerbatim("SimG4CoreApplication") << "OscarMTMasterThread::beginRun";
 
-  // Reading from ES must be done in the main (CMSSW) thread
-  readES(iSetup);
-
+  if (m_firstRun) {
+    if (m_pGeoFromDD4hep) {
+      m_pDD4Hep = &(*iSetup.getTransientHandle(m_DD4Hep));
+    } else {
+      m_pDDD = &(*iSetup.getTransientHandle(m_DDD));
+    }
+    m_pTable = &(*iSetup.getTransientHandle(m_PDT));
```

Comment:
Thi match what was done before, this should be
```C++
m_pTable =& iSet.getData(m_PDT);
```

as previously this was obtained via a `edm::Handle` not an `edm::TransientHandle`.

Reply:
Done